erpnext/hr/doctype/employee_transfer/employee_transfer.py

```diff
@@ -60,8 +79,88 @@ def on_cancel(self):
 			employee.company = self.company
 		employee.save()
 
+		# Run on_cancel hooks from other apps
+		for fn in frappe.get_hooks("employee_transfer_on_cancel"):
+			frappe.get_attr(fn)(self)
+
 	def validate_user_in_details(self):
 		for item in self.transfer_details:
 			if item.fieldname == "user_id" and item.new != item.current:
 				return True
 		return False
+
+	@frappe.whitelist()
+	def revert_temporary_transfer(self):
+		"""Revert temporary transfer by updating the existing record"""
+		if not self.is_temporary_transfer or not self.docstatus == 1:
+			return
+
+		if self.is_reverted:
+			frappe.throw(_("This transfer has already been reverted"))
+
+		# Get the employee document
+		employee = frappe.get_doc("Employee", self.employee)
+		
+		# Build the reversion details (swapping current/new values)
+		for detail in self.transfer_details:
+			fieldname = detail.fieldname or detail.property
+			
+			# Check if field exists in Employee doctype (either standard or custom)
+			field_exists = frappe.get_meta("Employee").has_field(fieldname)
+			
+			# If field doesn't exist in standard meta, check if it's a custom field or an attribute
+			if not field_exists:
+				if frappe.db.exists("Custom Field", {"dt": "Employee", "fieldname": fieldname}):
+					field_exists = True
+				elif hasattr(employee, fieldname):
+					field_exists = True
+			
+			if not field_exists:
+				continue
+			
+			# Update the transfer detail (reversing the values) using db_set
+			current_value = detail.new
+			new_value = detail.current
+			detail.db_set('current', current_value)
+			detail.db_set('new', new_value)
+
+		# Mark as reverted
+		self.db_set('is_reverted', 1)
+		
+		# Update the employee with the reverted values
+		for detail in self.transfer_details:
+			if detail.fieldname:
+				if hasattr(employee, 'db_set'):
+					employee.db_set(detail.fieldname, detail.new)
+				else:
```

Comment:
do not db_set the employee values, rather use the `.save()` to allow the controller to run it's validations